SourceCode/raytracer/raytracer/Animation.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+#include "CRTTypes.h"
+#include <memory>
+
+struct SceneObject;
+
+class Animation
+{
+public:
+    virtual void apply(SceneObject& obj) = 0;
+    auto getStartFrame() const { return startFrame; }
+    auto getEndFrame() const { return endFrame; }
+
+protected:
+    int startFrame = 0;
+    int endFrame = 0;
+    Animation(int startFrame, int endFrame) : startFrame(startFrame), endFrame(endFrame) {}
+};
+
+enum class MoveType
+{
+    Dolly,
+    Pedestal,
+    Truck
+};
+
+class MoveAnimation : public Animation 
+{
+
+public:
+    /*
+    * Will apply `delta` on every frame
+    */
+    MoveAnimation(const Vec3 delta, int startFrame, int endFrame) : Animation(startFrame, endFrame), delta(delta) {}
+
+    /*
+    * Will distribute movement over `endFrame - startFrame` frames
+    */
+    static std::shared_ptr<Animation> Make(MoveType type, float distance, int startFrame, int endFrame)
```

Comment:
Returning shared pointer here seems a bit unnecessary, may be you could try to return the Vector directly